### PR TITLE
libz blitz: use question mark instead of unwrap in examples

### PR DESCRIPTION
**Pull request by muzikmoe**
Rebase pull request #141 on `master`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -70,35 +70,59 @@
...
 //! use semver::Version;
+//! use std::error::Error;
 //!
-//! let mut bugfix_release = Version::parse("1.0.0").unwrap();
+//! # fn try_increment_patch() -> Result<(), Box<Error>> {
+//! let mut bugfix_release = Version::parse("1.0.0")?;
 //! bugfix_release.increment_patch();
 //!
 //! assert_eq!(Ok(bugfix_release), Version::parse("1.0.1"));
+//!
```
Comment (KodrAus) [inline src/lib.rs]: Let's remove these extra newlines before the hidden lines. Otherwise the docs will have blank lines at the end of the code blocks.

**Comment by muzikmoe** [comment]
Thank for your guiding.

File: src/lib.rs [hunk 2]
```diff
@@ -158,7 +186,8 @@
 //! ```
 
 #![doc(html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk-v2.png",
-       html_favicon_url = "https://www.rust-lang.org/favicon.ico")]
+      html_favicon_url = "https://www.rust-lang.org/favicon.ico")]
+#![doc(html_root_url = "https://docs.rs/semver/0.9.0")]
```
Comment (steveklabnik) [inline src/lib.rs]: this change looks unrelated?

File: src/lib.rs [hunk 3]
```diff
@@ -70,35 +70,56 @@
 //!
 //! ```{rust}
 //! use semver::Version;
+//! use std::error::Error;
```
Comment (KodrAus) [inline src/lib.rs]: Ah, sorry I didn't pick this up before! We should probably hide these `use std::error::Error` lines too since they don't actually relate to the example.

I would probably just remove that import altogether and change the function signatures to:

```rust
# fn try_increment_patch() -> Result<(), Box<::std::error::Error>> {
```

**Comment by opilar** [comment]
It's in conflict now. @muzikmoe if you're busy, I can resolve it for you.

**Comment by muzikmoe** [comment]
@opilar thank you.

**Comment by steveklabnik** [comment]
My comment about the unrelated change is still here, and it still needs a rebase. @muzikmoe any interest in keeping up with this PR?

**Comment by steveklabnik** [comment]
Triage: seems like this PR got lost in the shuffle; I'm going to give it a close. Thanks for the effort.
